### DISCUSSION TITLE
Getting data-position information on mouse click.

### DISCUSSION THREAD

**Discussion by plutorkhan** [google/model-viewer#4031] [Q&A]
I'd like to know how to get data-position (x, y, z) information when clicking on a specific part of the 3d model. This could be useful for adding annotations dynamically to specific parts of the model. 

A relevant example is present in the model-viewer editor, however, I couldn't find any methods or functions that return x, y, z coordinates for the mouse click.

**Comment by elalish** [comment]
This is the method you're looking for: https://modelviewer.dev/docs/#entrydocs-annotations-methods-positionAndNormalFromPoint

**Reply by plutorkhan** [reply]
Thanks a lot @elalish !

**Reply by aftabdivan-matrix** [reply]
How to get z coordinate.
